### PR TITLE
Move DotProd code into optimized affine layer

### PR DESCRIPTION
**Pull request by MinetaS**
This patch moves the DotProd code into the propagation function which has sequential access optimization. To prove the speedup, the comparison is done without the sparse layer. With the sparse layer the effect is marginal (GCC 0.3%, LLVM/Clang 0.1%).

For both tests, binary is compiled with GCC 14.1. Each test had 50 runs.

Sparse layer included:
```
speedup        = +0.0030
P(speedup > 0) =  1.0000
```

Sparse layer excluded:
```
speedup        = +0.0561
P(speedup > 0) =  1.0000
```

No functional change

### CODE REVIEW

**Comment by Torom** [comment]
Raspberry Pi 5 `armv8-dotprod`

clang 18.1.3
```
Result of  30 runs
==================
base (../stockfish   ) =     265956  +/- 543
test (...55217fd-5-PR) =     266660  +/- 444
diff                   =       +704  +/- 640

speedup        = +0.0026
P(speedup > 0) =  0.9843

CPU: 4 x aarch64
Hyperthreading: off
```

gcc 13.2.0
```
Result of  30 runs
==================
base (...tockfish_gcc) =     267766  +/- 460
test (...7fd-5-PR_gcc) =     268909  +/- 545
diff                   =      +1143  +/- 862

speedup        = +0.0043
P(speedup > 0) =  0.9953

CPU: 4 x aarch64
Hyperthreading: off
```

**Comment by MinetaS** [comment]
Fixed wrong commit description.

Also note: Neon without dotprod is faster in `affine_transform_non_ssse3` but I don't know the exact reason. Possibly `neon_m128_add_dpbusd_epi32` is slower than what we have now.

File: src/nnue/layers/simd.h [hunk 1]
```diff
@@ -99,9 +99,9 @@ namespace Stockfish::Simd {
 
 [[maybe_unused]] static void
 dotprod_m128_add_dpbusd_epi32(int32x4_t& acc, int8x16_t a, int8x16_t b) {
-
     acc = vdotq_s32(acc, a, b);
 }
+
```
Comment (Disservin) [inline src/nnue/layers/simd.h:104]: can you revert the change here please?

**Review by Disservin — COMMENTED** [review]
@Sopel97 anything to addd?

File: src/nnue/layers/simd.h (hunk 1 above)
Comment (MinetaS) [reply src/nnue/layers/simd.h]: Fixed. I test and modify code here and there so there might be other places I missed which break the formatting. In that case it's ok for you to correct them during the merge.

**Comment by Sopel97** [comment]
looks alright
